src/ompl/base/PlannerTerminationCondition.h

```diff
@@ -120,6 +121,32 @@ namespace ompl
 
         /** \brief Return a termination condition that will become true \e duration seconds in the future (wall-time), but is checked in a separate thread, every \e interval seconds; \e interval must be less than \e duration */
         PlannerTerminationCondition timedPlannerTerminationCondition(double duration, double interval);
+
+        /** \brief Return a termination condition that will become true as soon as the problem definition has an exact solution */
+        PlannerTerminationCondition exactSolnPlannerTerminationCondition(ompl::base::ProblemDefinitionPtr pdef);
```

Comment:
You can get cost-optimizing planners to terminate by setting the cost threshold in the optimization objective to infinite cost. You can also just the planner progress properties of the benchmark tools (written by @giogadi) to figure out the cost of the first solution as well as the convergence rate to the optimal solution (with a few sql queries to the benchmark database).